models/inherit.py

```diff
@@ -28,20 +30,45 @@ def current_user_rents(self):
         
     @api.multi
     def get_rent_count(self):
-        data_obj = self.env['bike.rent']
-        print("*****************************************")
-        print(data_obj.partner_id.name)
-        print("*****************************************")
-
         for record in self:
+            record._compute_rent_count = len(record.partner_ids)
+            print(record._compute_rent_count)
+
+class account_invoice(models.Model):
+    _inherit = 'account.invoice'
+    return_date = fields.Date(string='Bike return date')
+    
+    @api.multi
+    def return_date(self):
+        bike_rent_obj = self.env['sale.order']
+        self.return_date = bike_rent_obj.rent_stop
 
-            print("*****************************************")
-#            record._compute_rent_count = data_obj.search_count(['record.active', '=', True])
-            record._compute_rent_count = data_obj.search_count(['record.name', '=', data_obj.partner_id])
-#            list_data = data_obj.search(['record.name', '=', data_obj.partner_id])
-#            print(len(list_data))
-            print("*****************************************")
+class SaleOrder(models.Model):
+    _inherit = 'sale.order'
+  
+    rent_start = fields.Date(string='Rent start', required=True)
+    rent_stop = fields.Date(string='Rent stop', required=True)
+    rent_duration = fields.Integer(string='Rent duration')
     
-class Chatter(models.Model):
-    _name = 'user.chatter'
-    _inherit = 'mail.thread'
\ No newline at end of file
+    @api.multi
+    @api.depends('rent_duration')
+    def rent_dates(self):
+        for record in self:
+            record.rent_start = datetime.today().strftime('%Y-%m-%d')
+            record.rent_stop = record.rent_start + timedelta(record.rent_duration)
```

Comment:
1. `depends ` makes this a compute method, but fields are not declared as computed.
2. in `rent_start` field you assign a string value, in `rent_stop` a `datetime` object.
3. if I were to change `rent_duration`, let's say 5 days after the sale, `rent_start` would be reset to todays date.